index.js

```diff
@@ -1,5 +1,12 @@
 'use strict';
 
+const promisify = require('util').promisify;
+
+const supportsPromisify = typeof promisify === 'function';
+const canUseNativePromisify = opts => {
+	return supportsPromisify && opts.promiseModule === Promise && opts.errorFirst && !opts.multiArgs;
```

Comment:
You can do all these with `util.promisify` using https://nodejs.org/api/util.html#util_custom_promisified_functions.

Reply:
You can, but you don't gain the performance benefits of `util.promisify`. The underlying implementation would still be the pify method with a bit more setup required. 

**Current:**
```js
if (isCustom) {
     return getPifyImplementation(fn);
}
```

**Using custom:**
```js
if (isCustom) {
     const promiseFn = return getPifyImplementation(fn);
     fn[util.promisify.custom] = promiseFn;
     return promisify(fn);
}
```